app/sidekiq/benefits_intake_status_job.rb

```diff
@@ -21,10 +21,16 @@ def initialize(batch_size: BATCH_SIZE)
 
   def perform
     Rails.logger.info('BenefitsIntakeStatusJob started')
+    resolved_form_submissions = FormSubmission
+                                .joins(:form_submission_attempts)
+                                .where(form_submission_attempts: { aasm_state: %w[vbms failure] })
     pending_form_submissions = FormSubmission
                                .joins(:form_submission_attempts)
                                .where(form_submission_attempts: { aasm_state: 'pending' })
-    total_handled, result = batch_process(pending_form_submissions)
+    # We're calculating the resolved_form_submissions and subtracting them because it is possible for a FormSubmission
+    # to have two (or more) attempts, one 'pending' and the other 'vbms'. In such cases we don't want to include
+    # that FormSubmission because it has been resolved.
+    total_handled, result = batch_process(pending_form_submissions - resolved_form_submissions)
```

Comment:
If any ActiveRecord wizards can improve/condense this query, I'm all ears!!

Reply:
Avoid loading both sets of objects into memory. Filter the pending submissions directly in the database. For example:
```
form_submissions_and_attempts = FormSubmission.joins(:form_submission_attempts)

resolved_form_submission_ids = form_submissions_and_attempts
  .where(form_submission_attempts: { aasm_state: %w[vbms failure] })
  .pluck(:id)

pending_form_submissions = form_submissions_and_attempts
  .where(form_submission_attempts: { aasm_state: 'pending' })
  .where.not(id: resolved_form_submission_ids)

total_handled, result = batch_process(pending_form_submissions)
```

Reply:
Thanks @freeheeling ! Updated the branch.